posthog/temporal/common/clickhouse.py

```diff
@@ -599,9 +624,33 @@ async def acheck_query(
         elif "QueryStart" in events:
             return ClickHouseQueryStatus.RUNNING
         else:
-            self.logger.warning("Expected event not found in query log", query_id=query_id, events=events)
             raise ClickHouseQueryNotFound(query, query_id)
 
+    async def acheck_query_in_process_list(self, query_id: str) -> bool:
+        """Check if a query is running in the ClickHouse process list.
+
+        Arguments:
+            query_id: The ID of the query to check.
+
+        Returns:
+            True if the query is running, False otherwise.
+        """
+        query = """
+                SELECT count() > 0 AS is_running
```

Comment:
Why `count`? Would there ever be more than one row? Isn't it wasteful to do a `count` when just checking if there is one value would be enough?

Reply:
yeah good point, will update